test/boost/sstable_3_x_test.cc

```diff
@@ -3216,41 +3220,39 @@ static void compare_sstables(const std::filesystem::path& result_path, sstring t
     }
 }
 
-static tmpdir write_sstables(test_env& env, schema_ptr s, lw_shared_ptr<replica::memtable> mt1, lw_shared_ptr<replica::memtable> mt2, sstable_version_types version) {
-    tmpdir tmp;
-    auto sst = env.make_sstable(s, tmp.path().string(), 1, version, sstable::format_types::big, 4096);
+static void compare_sstables(test_env& env, sstring table_name, sstable_version_types version) {
+    compare_sstables(env.tempdir().path().native(), std::move(table_name), version);
+}
+
+static void write_sstables(test_env& env, schema_ptr s, lw_shared_ptr<replica::memtable> mt1, lw_shared_ptr<replica::memtable> mt2, sstable_version_types version) {
+    auto sst = env.make_sstable(s, env.tempdir().path().string(), 1, version, sstable::format_types::big, 4096);
```

Comment:
sstables::test_env() has a dir-less overload that uses its tempdir transparently